datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -67,6 +68,9 @@ public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<In
 
     LOG.info("old partition assignment info, assignment: {}", currentAssignment);
 
+    Validate.isTrue(currentAssignment.size() > 0,
```

Comment:
Isn't this going to raise IllegalArgumentException since for whatever reason we are seeing it be 0? Not sure I understand what the change set out to achieve. Throw a more appropriate exception?

Reply:
I feel IllegalArgumentException is appropriate for this api. This change is to not do the assignment if total assignments are zero and redo leader assignment.

Reply:
Just to clarify, the behavior of this this won't change right, just the type of exception thrown? Even before this change, an exception would be thrown and the assignment retried?

Reply:
Yes, Just Arithmetic Exception is something we want to avoid.